sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,299 @@
+-- Query first observations for Firefox Android Clients.
+WITH first_seen AS (
+  SELECT
+    client_id,
+    first_seen_date,
+    country AS first_reported_country,
+    isp AS first_reported_isp,
+    DATETIME(first_run_date) AS first_run_datetime,
+    normalized_channel AS channel,
+    device_manufacturer,
+    device_model,
+    normalized_os_version AS os_version
+  FROM
+    `mozdata.fenix.baseline_clients_first_seen`
+  WHERE
+    submission_date = @submission_date
+    AND normalized_channel = 'release'
+),
+-- Find earliest data per client from the first_session ping.
+first_session_ping AS (
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(SAFE.PARSE_DATETIME('%F', SUBSTR(client_info.first_run_date, 1, 10))) AS first_run_datetime,
+    ARRAY_AGG(
+      metrics.string.first_session_campaign
+      ORDER BY
+        metrics.datetime.first_session_timestamp ASC
+    )[SAFE_OFFSET(0)] AS adjust_campaign,
+    ARRAY_AGG(
+      metrics.string.first_session_network
+      ORDER BY
+        metrics.datetime.first_session_timestamp ASC
+    )[SAFE_OFFSET(0)] AS adjust_network,
+    ARRAY_AGG(
+      metrics.string.first_session_adgroup
+      ORDER BY
+        metrics.datetime.first_session_timestamp ASC
+    )[SAFE_OFFSET(0)] AS adjust_ad_group,
+    ARRAY_AGG(
+      metrics.string.first_session_creative
+      ORDER BY
+        metrics.datetime.first_session_timestamp ASC
+    )[SAFE_OFFSET(0)] AS adjust_creative
+  FROM
+    `mozdata.fenix.first_session`
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND SAFE.PARSE_DATE('%F', SUBSTR(client_info.first_run_date, 1, 10)) >= DATE(
+      submission_timestamp
+    ) -- first_session ping received after the client is first seen.
+    AND ping_info.seq = 0 -- Pings are sent in sequence, this guarantees that the first one is returned.
```

Comment:
If we're doing the array_agg ordering above, do we need this?

Reply:
The submission_timestamp is when the ping reaches the server and the seq is the order in which pings are generated. Not necessarily the pings are received in the server in the same order they were generated, so this guarantees that we use the first one, is important to keep it. However the OFFSET adds a level of safety in case more than one ping is generated with the same seq (shouldn't happen tho)